src/utils.ts

```diff
@@ -107,10 +107,56 @@ export async function getBalance(
   return value;
 }
 
-export function uiToNative(amount: number, decimals: number): BN {
-  return new BN(Math.round(amount * Math.pow(10, decimals)));
+export function nativeExp10(exponent: number) {
+  if (exponent < 0) {
+    throw new Error(
+      'BN does not support negative exponents (fractional values)'
+    );
+  }
+  return new BN('1' + '0'.repeat(exponent));
```

Comment:
What is the targetted maximum exponent here?

`new BN(10 ** exponent)` should be enough

<img width="587" alt="Screenshot 2022-12-29 at 16 36 02" src="https://user-images.githubusercontent.com/22965416/209932543-fa00a40f-d1bf-4790-ad12-5afc550af2e2.png">

Reply:
I think there is not real bounds here, it could be 100 for what we know

Reply:
IMO, the bound is our usecase. `uxd-client` is made to work for solana. In solana, the maximum token decimals size we have encountered/worked with is 9 decimals _(UXP)_.

Reply:
I see your point, ima limit the use of those functions to solana stuff only then 👌